MarsRover.Tests/MapShouldTestData.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace MarsRover.Tests
+{
+    public class MapShouldTestData
+    {
+        public static TheoryData<List<Square>, int, int, Location, SquareContent> MapTestData =
+            new TheoryData<List<Square>, int, int, Location, SquareContent>
+            {
+                {
+                    new List<Square>
+                    {
+                        new Square(SquareContent.Obstacle, 0, 0),
+                        new Square(SquareContent.None, 1, 0),
+                        new Square(SquareContent.None, 2, 0),
+                        new Square(SquareContent.None, 3, 0),
+                        new Square(SquareContent.None, 0, 1),
+                        new Square(SquareContent.None, 1, 1),
+                        new Square(SquareContent.None, 2, 1),
+                        new Square(SquareContent.None, 3, 1),
+                        new Square(SquareContent.None, 0, 2),
+                        new Square(SquareContent.None, 1, 2),
+                        new Square(SquareContent.None, 2, 2),
+                        new Square(SquareContent.None, 3, 2)

```

Comment:
I know you've probably learned that comments are often bad, but here it could make sense to have a comment to show you more easily what this test data is
```
ONNN
NNNN
NNNN
```